thorclient/thorclient.go

```diff
@@ -0,0 +1,203 @@
+)
+
+type Client struct {
+	httpConn *httpclient.Client
+	wsConn   *wsclient.Client
+}
+
+func New(url string) *Client {
+	return &Client{
+		httpConn: httpclient.NewClient(url),
+	}
+}
+
+func NewWithWS(url string) (*Client, error) {
+	wsClient, err := wsclient.NewClient(url)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		httpConn: httpclient.NewClient(url),
+		wsConn:   wsClient,
+	}, nil
+}
+
+func (c *Client) RawHTTPClient() *httpclient.Client {
+	return c.httpConn
+}
+func (c *Client) RawWSClient() *wsclient.Client {
+	return c.wsConn
+}
+
+func (c *Client) GetTransactionReceipt(id *thor.Bytes32) (*transactions.Receipt, error) {
+	return c.httpConn.GetTransactionReceipt(id)
+}
+
+func (c *Client) InspectClauses(calldata *accounts.BatchCallData) ([]*accounts.CallResult, error) {
+	return c.httpConn.InspectClauses(calldata)
+}
+
+func (c *Client) InspectTxClauses(tx *tx.Transaction, senderAddr *thor.Address) ([]*accounts.CallResult, error) {
+	clauses := convertToBatchCallData(tx, senderAddr)
+	return c.InspectClauses(clauses)
+}
+
+func (c *Client) SendTransaction(tx *tx.Transaction) (*common.TxSendResult, error) {
+	rlpTx, err := rlp.EncodeToBytes(tx)
+	if err != nil {
+		return nil, fmt.Errorf("unable to encode transaction - %w", err)
+	}
+
+	return c.SendTransactionRaw(rlpTx)
+}
+
+func (c *Client) SendTransactionRaw(rlpTx []byte) (*common.TxSendResult, error) {
+	return c.httpConn.SendTransaction(&transactions.RawTx{Raw: hexutil.Encode(rlpTx)})
+}
+
+func (c *Client) FilterEvents(req *events.EventFilter) ([]events.FilteredEvent, error) {
+	return c.httpConn.FilterEvents(req)
+}
+
+func (c *Client) FilterTransfers(req *events.EventFilter) ([]*transfers.FilteredTransfer, error) {
+	return c.httpConn.FilterTransfers(req)
+}
+
+func (c *Client) GetAccount(addr *thor.Address) (*accounts.Account, error) {
+	return c.httpConn.GetAccount(addr, "")
+}
+
+func (c *Client) GetAccountForRevision(addr *thor.Address, revision string) (*accounts.Account, error) {
+	return c.httpConn.GetAccount(addr, revision)
+}
+
+func (c *Client) GetAccountCode(addr *thor.Address) ([]byte, error) {
+	return c.httpConn.GetAccountCode(addr, "")
+}
+
+func (c *Client) GetAccountCodeForRevision(addr *thor.Address, revision string) ([]byte, error) {
+	return c.httpConn.GetAccountCode(addr, revision)
+}
+
+func (c *Client) GetStorage(addr *thor.Address, key *thor.Bytes32) ([]byte, error) {
+	return c.httpConn.GetStorage(addr, key)
+}
+
+func (c *Client) GetBlockExpanded(block string) (blocks *blocks.JSONExpandedBlock, err error) {
+	return c.httpConn.GetBlockExpanded(block)
+}
+
+func (c *Client) GetBlock(block string) (blocks *blocks.JSONBlockSummary, err error) {
+	return c.httpConn.GetBlock(block)
+}
+
+func (c *Client) GetTransaction(id *thor.Bytes32) (*transactions.Transaction, error) {
+	return c.httpConn.GetTransaction(id, false)
+}
+
+func (c *Client) GetTransactionPending(id thor.Bytes32) (*transactions.Transaction, error) {
```

Comment:
We have inconsistent parameters.

Pointer: 
```golang
GetTransaction(id *thor.Bytes32)
```


Non pointer:
```golang
GetTransactionPending(id thor.Bytes32)
```

`transaction.ID()` returns a non pointer, so I would be inclined to use that:

```golang
func (t *Transaction) ID() (id thor.Bytes32)
```

Reply:
Good point, fixed ! 

> transaction.ID() returns a non pointer, so I would be inclined to use that:
> func (t *Transaction) ID() (id thor.Bytes32)

I think the returning `transactions.Transaction` doesn't have that method, but it does have the ID field. Given that's it's lighter to pass around pointer, it's probably worth defaulting to `id *thor.Bytes32`? Wdyt ?